src/conky.cc

```diff
@@ -742,16 +742,24 @@ int percent_print(char *buf, int size, unsigned value) {
  * The algorithm always divides by 1024, as unit-conversion of byte
  * counts suggests. But for output length determination we need to
  * compare with 1000 here, as we print in decimal form. */
-void human_readable(long long num, char *buf, int size) {
+void human_readable(int use_base, const char *s, long long num, char *buf, int size) {
```

Comment:
Actually, this is a bit confusing. Let's just have 2 separate functions: one which applies the base multiplier depending on whether or not the object has the `si` argument, and then we can leave `human_readable` as-is. Then you can call it like:

```C++
human_readable(apply_base_multiplier(obj->data.s, info.cached), p, p_max_size);
```

Make sense?

Reply:
Done :heart_decoration: